src/bellmanford.jl

```diff
@@ -105,3 +108,28 @@ function has_negative_edge_cycle{V, D}(
     edge_inspector = VectorEdgePropertyInspector{D}(edge_dists)
     has_negative_edge_cycle(graph, edge_inspector)
 end
+
+function enumerate_paths{V,D<:Number}(state::BellmanFordStates{V,D}, dest::Vector{V})
+    parents = state.parents
+    hasparent = state.hasparent
+    
+    num_dest = length(dest)
+    all_paths = Array(Vector{V},num_dest)
+    for i=1:num_dest
+        all_paths[i] = V[]
+        index = dest[i]
+        if hasparent[index] || parents[index] == index
+            while hasparent[index]
```

Comment:
My understanding is that in most cases, we do in fact only care about enumerating paths to a small subset (or even just one) vertex in the Graph (e.g. [`extractRoute` in OpenStreetMap](https://github.com/tedsteiner/OpenStreetMap.jl/blob/master/src/routing.jl#L211-L229)). It might also be unclear what admissible heuristic to use for A* search (although [PRs are always welcome!](https://github.com/tedsteiner/OpenStreetMap.jl/issues/17)).

To state just one application: in geospatial analysis, you might have a transportation network, where you care about travelling from say.. an emergency location to nearby hospitals (and then ask for the respective routes); or you could be finding the location for a warehouse, and the routes to the different distributors.